### PR TITLE
Remove the mysql jar from the jdbc connector

### LINKED ISSUE #6656: The mysql jar used by the jdbc connector has an invalid license.
**Describe the bug**
In 2.5.0 release the jdbc connector packages mysql-connector-java:jar:8.0.11. This jar has no valid license. It is GPLV2 with a deprecated FOSS exception

Because this FOSS exception no longer exists, the license is rendered invalid.

### PR DESCRIPTION
**Pull request by zymap**
---

Fixes #6656

*Motivation*:

Remove the mysql jar from the jdbc connector

### CODE REVIEW

**Comment by sijie** [comment]
@zymap when you removed mysql from jdbc connector, do you need to fix the integration test?

**Comment by zymap** [comment]
Let me take a look.

File: tests/integration/src/test/java/org/apache/pulsar/tests/integration/functions/PulsarFunctionsTest.java [hunk 1]
```diff
@@ -123,7 +123,6 @@ public void testHdfsSink() throws Exception {
         testSink(new HdfsSinkTester(), false);
     }
 
-    @Test(groups = "sink")
```
Comment (sijie) [inline tests/integration/src/test/java/org/apache/pulsar/tests/integration/functions/PulsarFunctionsTest.java:126]: I don't think we should disable the test for JDBC connector.

**Comment by vzhikserg** [comment]
@sijie @zymap  the pull request #6835 removed the MySQL connector.

**Comment by sijie** [comment]
Since #6835 already removed MySQL, close this issue now.

**Pull request closed without merging**